sw/device/tests/BUILD

```diff
@@ -1303,7 +1303,7 @@ opentitan_functest(
     srcs = ["sram_ctrl_execution_test.c"],
     deps = [
         "//hw/top_earlgrey/sw/autogen:top_earlgrey",
-        "//sw/device/examples/sram_program:sram_program_ar",
+        "//sw/device/examples/sram_program",
```

Comment:
@alphan lmk what you think, but I thought it might be easier for folks not to have to embed macro generated dep labels into other rules as this requires one to look at the Bazel macro code to know what these names should be.